src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // делим дату на части
+  const dateParts = date.split(fromFormat[3]);
+  const newDateObject = {};
+
+  for (let i = 0; i < 3; i++) {
+    newDateObject[fromFormat[i]] = dateParts[i];
+  }
+  // проверка если в fromFormat есть YYYY, а в toFormat — YY,
+  // то newDateObject['YY'] будет присвоено последние две цифры
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    newDateObject['YY'] = newDateObject['YYYY'].slice(-2);
+  } else if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = parseInt(newDateObject['YY']);
+
+    newDateObject['YYYY'] =
+      year < 30 ? '20' + newDateObject['YY'] : '19' + newDateObject['YY'];
```

Comment:
The conversion from 'YYYY' to 'YY' and from 'YY' to 'YYYY' is correctly implemented, but it's a good practice to avoid magic numbers. Consider defining constants for the thresholds like `const CURRENT_CENTURY_THRESHOLD = 30;` and `const CENTURY_1900 = '19';` and `const CENTURY_2000 = '20';` to make the code more readable and maintainable.